db/crud.go

```diff
@@ -1921,6 +1921,20 @@ func (db *Database) checkDocChannelsAndGrantsLimits(docID string, channels base.
 			base.WarnfCtx(db.Ctx, "Doc id: %v access and role grants count: %d exceeds %d for grants per doc warning threshold", base.UD(docID), grantCount, *grantThreshold)
 		}
 	}
+
+	// Warn when channel names are larger than a configured threshold
+	if channelNameSizeThreshold := db.Options.UnsupportedOptions.WarningThresholds.ChannelNameSize; channelNameSizeThreshold != nil {
+		var channelsOverCount int
+		for c := range channels {
+			if uint32(len(c)) > *channelNameSizeThreshold {
+				channelsOverCount++
+			}
+		}
+		if channelsOverCount > 0 {
+			db.DbStats.Database().WarnChannelNameSizeCount.Add(1)
+			base.WarnfCtx(db.Ctx, "Doc id: %v channel name size: %d channel/s exceeds %d characters for channel name size warning threshold", base.UD(docID), channelsOverCount, *channelNameSizeThreshold)
```

Comment:
I think it'd be nice to include the channel name / names that exceed the limit so a customer doesn't have to go and inspect the doc / xattrs and count themselves.

Hwoever, also thinking that potentially logging multiple > 250 char strings in a long messages might be a bit excessive. Maybe have a play around with it? Perhaps we could even log once per channel thats too large inside the loop?